src/renderer/components/virtual-list/virtual-list.tsx

```diff
@@ -126,8 +126,7 @@ interface RowProps extends ListChildComponentProps {
 const Row = observer((props: RowProps) => {
   const { index, style, data } = props;
   const { items, getRow } = data;
-  const item = items[index];
-  const uid = typeof item == "string" ? index : items[index].getId();
+  const uid = items[index].getId ? items[index].getId() : index;
```

Comment:
How you can be sure you're that you are checking `getId` from object when you defined items as `any: []`?
```suggestion
  const uid = items[index]?.getId() ?? index;
```

Reply:
If `getId` is optionally present, shouldn't it be the following?
```suggestion
  const uid = items[index]?.getId?.() ?? index;
```

Reply:
The original code is (also) correct. getId will return undefined if it's not a property of the object, which is falsy, so indecx would be used in that case. I don't see how null coalescing operator is more readable than the conditional operator. (I think they're equivalent readability wise).